src/pages/user/FlavorStash.js

```diff
@@ -1,63 +1,347 @@
-import dayjs from 'dayjs';
 import PropTypes from 'prop-types';
-import { Helmet } from 'react-helmet';
-import React, { Component } from 'react';
-import { Container, Table } from 'react-bootstrap';
 import { connect } from 'react-redux';
+import { Helmet } from 'react-helmet';
+import React, { Component, Fragment } from 'react';
+import { Link } from 'react-router-dom';
 import { bindActionCreators } from 'redux';
-
-import { getStash, isLoaded } from 'selectors/flavor';
+import { Form as FinalForm, Field } from 'react-final-form';
+import {
+  Button,
+  ButtonGroup,
+  Card,
+  Col,
+  Container,
+  Form,
+  InputGroup,
+  Row
+} from 'react-bootstrap';
+import ToggleButton from 'components/ToggleButton/ToggleButton';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { actions as flavorActions } from 'reducers/flavor';
+import { getStash, isLoaded } from 'selectors/flavor';
+import Note from 'components/FlavorStash/Note';
 
 export class FlavorStash extends Component {
   static propTypes = {
     actions: PropTypes.shape({
-      requestStash: PropTypes.func.isRequired
-    }).isRequired,
-    stash: PropTypes.arrayOf(PropTypes.object),
-    stashLoaded: PropTypes.bool.isRequired
+      addStash: PropTypes.func.isRequired,
+      removeStash: PropTypes.func.isRequired,
+      requestStash: PropTypes.func.isRequired,
+      updateStash: PropTypes.func.isRequired
+    }),
+    stash: PropTypes.array.isRequired,
+    stashLoaded: PropTypes.bool
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      expanded: {},
+      editingStash: false,
+      removed: {},
+      usage: {},
+      viewingNote: {}
+    };
+
+    this.handleRemoveFromStash = this.removeFromStash.bind(this);
+    this.handleAddToStash = this.addToStash.bind(this);
+    this.handleStashEditor = this.handleStashEditor.bind(this);
+    this.handleExpandFlavor = this.expandFlavor.bind(this);
+    this.handleStashSubmit = this.handleStashSubmit.bind(this);
+  }
+
   componentDidMount() {
-    const { actions, stashLoaded } = this.props;
+    const { stashLoaded, actions } = this.props;
 
     if (!stashLoaded) {
       actions.requestStash();
     }
   }
 
+  date(d) {
+    const s = new Date(d);
+
+    return s.toLocaleDateString();
```

Comment:
Use date-fns `format` for this

Reply:
This seems to be something we should probably work on and integrate later. date-fns doesn't seem to be flexible enough to be used within the scope of this PR.